### PR TITLE
Support Self-Signed CA Cert and skip-cert-verify with Hub Client

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it

- The Hub Client will use the `GetCert` API to get the certificate data for the specified hub endpoint.
- If the certificate data is not found, it will skip the TLS configuration and let the client use default configuration.
- GetCert API now accepts URI along with hostname

### Which issue(s) this PR fixes

Fixes #

### Release note

```release-note
Support Self-Signed CA Cert and skip-cert-verify with Hub Client
GetCert API now accepts URI along with hostname
```

### CODE REVIEW

File: config/certs_test.go [hunk 1]
```diff
@@ -43,6 +43,14 @@ func TestSetGetDeleteCerts(t *testing.T) {
 	assert.Nil(t, err)
 	assert.Equal(t, cert1, ctx)
 
+	ctx, err = GetCert("https://test1")
```
Comment (prkalle) [inline config/certs_test.go:46]: nit: This was already existing, but can we change the `ctx` getCert` something so that it is more opt?

**Review by prkalle — APPROVED** [review]
LGTM. I have a minor nit comment. Thanks

File: config/certs_test.go (hunk 1 above)
Comment (vuil) [reply config/certs_test.go:46]: +1 
ctx->cert throughout would have made more sense.

File: config/certs.go [hunk 2]
```diff
@@ -30,6 +31,11 @@ func GetCert(host string) (*configtypes.Cert, error) {
 	if host == "" {
```
Comment (vuil) [inline config/certs.go:31]: suggestion: update comment to say "by host or URI"

File: config/certs_test.go [hunk 3]
```diff
@@ -43,6 +43,14 @@ func TestSetGetDeleteCerts(t *testing.T) {
...
+	assert.Nil(t, err)
+	assert.Equal(t, cert1, ctx)
+
+	ctx, err = GetCert("https://test1/fake")
```
Comment (vuil) [inline config/certs_test.go:50]: suggestion: add 3 more cases for 
ws://test1/fake and 
wss://test1/fake to show this works for websocket URIs as well.

and 
https://test1:12345/fake

File: config/certs_test.go [hunk 4]
```diff
@@ -43,6 +43,14 @@ func TestSetGetDeleteCerts(t *testing.T) {
...
+	assert.Nil(t, err)
+	assert.Equal(t, cert1, ctx)
+
```
Comment (vuil) [inline config/certs_test.go:53]: nit: suggest add another case of invalid uri 
ctx, err = GetCert("https://tes  t1/fak e") or something, should we choose to return a better error as suggested above.

File: config/certs.go [hunk 5]
```diff
@@ -30,6 +31,11 @@ func GetCert(host string) (*configtypes.Cert, error) {
...
 		return nil, errors.New("host is empty")
 	}
+	u, err := url.Parse(host)
```
Comment (vuil) [inline config/certs.go:34]: an unparsable URI ends up getting the same error as a missing entry. 
how about just return here,
with errors.New("invalid uri") or wrap the err?

**Review by vuil — COMMENTED** [review]
very nice!
Some nits and suggestion for additional test cases.

**Comment by vuil** [comment]
another nit:
Maybe GetCert deserves a mention in docs/config.md?

File: config/certs_test.go (hunk 4 above)
Comment (anujc25) [reply config/certs_test.go:53]: Added this case but I would prefer not to validate the URI/HostName as part of this API and just use the comparison with the existing cert store configuration and return the result.

**Review by vuil — APPROVED** [review]
lgtm, thanks

**Review by vuil — APPROVED** [review]

**Pull request merged**